pkgs/by-name/im/implot/package.nix

```
@@ -0,0 +1,41 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  callPackage,
+  cmake,
+  vcpkg,
+  imgui,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "implot";
+  version = "0.16";
+
+  src = fetchFromGitHub {
+    owner = "epezent";
+    repo = "implot";
+    rev = "v${version}";
+    hash = "sha256-/wkVsgz3wiUVZBCgRl2iDD6GWb+AoHN+u0aeqHHgem0=";
+  };
+
+  cmakeRules = "${vcpkg.src}/ports/implot";
+  postPatch = ''
+    cp "$cmakeRules"/CMakeLists.txt ./
```

```suggestion
  postPatch = ''
    cp "${vcpkg.src}/ports/implot"/CMakeLists.txt ./
```
or is this used as an env?

---

This is so as to keep phases agnostic of the hashes. I see no good reason to render the hashes into the scripts

---

I can't follow you.

Usually turning nix variables into bash variables is an anti pattern we try to avoid.

---

Introducing the auxiliary variable like this gives the opportunity to `implot.overrideAttrs (_: { cmakeRules = ./my-custom-cmake; })` which otherwise would require a careful surgery on the `postPatch`. I've been following a similar pattern outside the tree and thought it wouldn't do harm to do the same here.

>  an anti pattern we try to avoid.

Yesssss setup-hooks are an anti-pattern, let's make nix evaluation faster and abolish setup hooks 😆

---

You should be able to use finalAttrs to allow overrides like that, too.

---

That's an option, but it's not really any better than eliminating the template rendering and treating `postPatch` as a normal setup hook